### DISCUSSION TITLE
CORS OPTIONS(Preflight request) request should return 204

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [Kong/kong#7318] [Ideas and feature requests]
Next line returns 200 as status https://github.com/Kong/kong/blob/c817fada7a514d84f6e0aab7461605c71bf3a841/kong/plugins/cors/handler.lua#L13

Internet says it should return 204:
- https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request
- https://docs.microsoft.com/en-us/aspnet/core/security/cors?view=aspnetcore-3.1#preflight-requests

Internet changed its mind since 2018 https://github.com/Kong/kong/pull/4029

**Comment by ms2008** [comment]
The spec doesn’t prescribe what status code to return. See https://stackoverflow.com/a/46028619 for details.
